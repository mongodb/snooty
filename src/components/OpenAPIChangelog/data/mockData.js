export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
