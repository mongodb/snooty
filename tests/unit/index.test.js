import React from 'react';
import { render } from 'enzyme';
import Index from '../../src/templates/guides-index';

// data for this component
import mockData from './data/index.test.json';
import mockPageMetadata from './data/guidesPageMetadata.json';

jest.mock('../../src/hooks/use-site-metadata');

describe('Landing Page', () => {
  // TODO: DOCSP-6123: Handle fixing this test. Skip it for now.
  it.skip('renders correctly', () => {
    const tree = render(<Index pageContext={{ __refDocMapping: mockData, pageMetadata: mockPageMetadata }} />);
    expect(tree).toMatchSnapshot();
  });
});