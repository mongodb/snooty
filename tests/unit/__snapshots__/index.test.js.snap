// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page renders correctly 1`] = `
<div
  class="content"
>
  <div
    class="guide-category-list"
  >
    <div
      class="section"
      id="guides"
    >
      <h1>
        Guides
        <a
          class="headerlink"
          href="#guides"
          title="Permalink to this headline"
        >
          Â¶
        </a>
      </h1>
      <section
        class="guide-category"
      >
        <div
          class="guide-category__title guide-category__title--getting-started"
        >
          Getting Started
        </div>
        <div
          class="guide-category__guides"
        >
          <div
            class="guide-column"
          >
            <a
              class="guide guide--regular"
              href="server/introduction"
            >
              <div
                class="guide__title"
              >
                Structure your Data for MongoDB
              </div>
              <div
                class="guide__body"
              />
              <div
                class="guide__time"
              >
                15min
              </div>
            </a>
            <div
              class="guide guide--jumbo guide--expanded"
            >
              <div
                class="guide__title"
              >
                MongoDB in the Cloud
              </div>
              <ul
                class="guide__body"
              >
                <li
                  class="guide__entry"
                >
                  <a
                    href="cloud/atlas"
                  >
                    Create an Atlas Account and Cluster
                  </a>
                </li>
                <li
                  class="guide__entry"
                >
                  <a
                    href="cloud/connectionstring"
                  >
                    Set Up Atlas Connectivity
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="guide guide--jumbo guide--expanded"
            >
              <div
                class="guide__title"
              >
                CRUD Guides: Create, Read, Update, and Delete Data
              </div>
              <ul
                class="guide__body"
              >
                <li
                  class="guide__entry"
                >
                  <a
                    href="server/insert"
                  >
                    Insert Data into MongoDB
                  </a>
                </li>
                <li
                  class="guide__entry"
                >
                  <a
                    href="server/read"
                  >
                    Read Data from MongoDB
                  </a>
                </li>
                <li
                  class="guide__entry"
                >
                  <a
                    href="server/read_queries"
                  >
                    Read Data from MongoDB With Queries
                  </a>
                </li>
                <li
                  class="guide__entry"
                >
                  <a
                    href="server/read_operators"
                  >
                    Read Data using Operators and Compound Queries
                  </a>
                </li>
                <li
                  class="guide__entry"
                >
                  <a
                    href="server/update"
                  >
                    Update Data in MongoDB
                  </a>
                </li>
                <li
                  class="guide__entry"
                >
                  <a
                    href="server/delete"
                  >
                    Delete Data from MongoDB
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="guide-column"
          >
            <a
              class="guide guide--regular"
              href="server/install"
            >
              <div
                class="guide__title"
              >
                Install MongoDB
              </div>
              <div
                class="guide__body"
              />
              <div
                class="guide__time"
              >
                10min
              </div>
            </a>
            <div
              class="guide guide--jumbo guide--expanded"
            >
              <div
                class="guide__title"
              >
                Migrate to MongoDB Atlas
              </div>
              <ul
                class="guide__body"
              >
                <li
                  class="guide__entry"
                >
                  <a
                    href="cloud/migrate-from-aws-to-atlas"
                  >
                    Migrate a MongoDB Replica Set from AWS to MongoDB Atlas
                  </a>
                </li>
                <li
                  class="guide__entry"
                >
                  <a
                    href="cloud/migrate-from-compose"
                  >
                    Migrate from Compose to MongoDB Atlas
                  </a>
                </li>
              </ul>
            </div>
            <a
              class="guide guide--regular"
              href="stitch/react_googleauth"
            >
              <div
                class="guide__title"
              >
                Sample App: React + Stitch + Google Authentication
              </div>
              <div
                class="guide__body"
              />
              <div
                class="guide__time"
              >
                15min
              </div>
            </a>
          </div>
          <div
            class="guide-column"
          >
            <a
              class="guide guide--regular"
              href="server/auth"
            >
              <div
                class="guide__title"
              >
                Secure your MongoDB Deployment
              </div>
              <div
                class="guide__body"
              />
              <div
                class="guide__time"
              >
                10min
              </div>
            </a>
            <a
              class="guide guide--regular"
              href="server/import"
            >
              <div
                class="guide__title"
              >
                Import Data into MongoDB
              </div>
              <div
                class="guide__body"
              />
              <div
                class="guide__time"
              >
                15min
              </div>
            </a>
            <a
              class="guide guide--regular"
              href="server/drivers"
            >
              <div
                class="guide__title"
              >
                Connect to MongoDB
              </div>
              <div
                class="guide__body"
              />
              <ul
                class="guide__pills "
                role="tablist"
              >
                <li
                  data-tabid="shell"
                >
                  <span
                    class="guide__pill   guide__pill"
                    role="button"
                    tabindex="0"
                  >
                    Mongo Shell
                  </span>
                </li>
                <li
                  data-tabid="compass"
                >
                  <span
                    class="guide__pill   guide__pill"
                    role="button"
                    tabindex="1"
                  >
                    Compass
                  </span>
                </li>
                <li
                  data-tabid="python"
                >
                  <span
                    class="guide__pill   guide__pill"
                    role="button"
                    tabindex="2"
                  >
                    Python
                  </span>
                </li>
                <li
                  data-tabid="java-sync"
                >
                  <span
                    class="guide__pill   guide__pill"
                    role="button"
                    tabindex="3"
                  >
                    Java (Sync)
                  </span>
                </li>
                <li
                  class="guide__pill guide__pill--seeall"
                >
                  See All
                </li>
              </ul>
              <div
                class="guide__time"
              >
                15min
              </div>
            </a>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
